nodeup/pkg/model/context.go

```diff
@@ -268,3 +269,55 @@ func (c *NodeupModelContext) KubectlPath() string {
 	}
 	return kubeletCommand
 }
+
+// BuildCertificateTask is responsible for build a certificate request task
+func (c *NodeupModelContext) BuildCertificateTask(ctx *fi.ModelBuilderContext, name, filename string) error {
+	cert, err := c.KeyStore.FindCert(name)
+	if err != nil {
+		return err
+	}
+
+	if cert == nil {
+		return fmt.Errorf("certificate %q not found", name)
+	}
+
+	serialized, err := cert.AsString()
+	if err != nil {
+		return err
+	}
+
+	ctx.AddTask(&nodetasks.File{
+		Path:     filepath.Join(c.PathSrvKubernetes(), filename),
+		Contents: fi.NewStringResource(serialized),
+		Type:     nodetasks.FileType_File,
+		Mode:     s("0400"),
+	})
+
+	return nil
+}
+
+// buildPrivateKeyTask is responsible for build a certificate request task
+func (c *NodeupModelContext) BuildPrivateTask(ctx *fi.ModelBuilderContext, name, filename string) error {
```

Comment:
Function name differs from associated comment, just a minor correction to comment